### PR TITLE
Default template configuration for ITIL objects

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

**Comment by trasher** [comment]
This fix issue at storage time. An improvement would be to make sur a default template is loaded on the initial form, to mark the content field as required in dom.

File: inc/commonitilobject.class.php [hunk 1]
```diff
@@ -7561,6 +7561,9 @@ function getITILTemplateToUse(
       $itilcategories_id = 0,
       $entities_id = -1
    ) {
+      if (!$type && $this->getType() != Ticket::getType()) {
```
Comment (cedric-anne) [inline inc/commonitilobject.class.php]: I do not understand why `$this->getType() != Ticket::getType()` as we previously check for `if ($this->getType() == Ticket::getType()) {`.

Comment (cedric-anne) [reply inc/commonitilobject.class.php]: Got it.

File: tests/functionnal/ITILTemplate.php [hunk 2]
```diff
@@ -410,4 +410,102 @@ public function testTasks($itiltype) {
...
+      $profile_tpl_id = $this->createTemplate($itiltype);
+      $profile = getItemByTypeName('Profile', 'Technician');
+      $this->boolean($profile->update(['id' => $profile->fields['id'], $field => $profile_tpl_id]))->isTrue();
+
+      //login again to refresh profile
+      $this->login('tech', 'tech');
+
+      $tt = $itilobject->getITILTemplateToUse(0, $type, 0);
+      $this->boolean($tt->isNewItem())
+           ->isFalse('Not template expected from pofile assignment');
```
Comment (cedric-anne) [inline tests/functionnal/ITILTemplate.php]:
```suggestion
           ->isFalse('Not template expected from profile assignment');
```

**Pull request merged**